clusterversion: move proto to separate package

RaduBerinde opened the pull request:
This change moves the `ClusterVersion` proto to a separate `clusterversionpb` package. The `Settings` code will be able to use it from here and store `ClusterVersion` directly in the in-memory settings.

Informs: #113385
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/113976)
<!-- Reviewable:end -->

RaduBerinde committed:
clusterversion: move proto to separate package

This change moves the `ClusterVersion` proto to a separate
`clusterversionpb` package. The `Settings` code will be able to use it
from here and store `ClusterVersion` directly in the in-memory
settings.

We move `IsActive` from a method on the `ClusterVersion` to a method
on the `clusterverison.Key` (which makes more sense anyway, we are
asking if the key is active).

Informs: #113385
Release note: None
